repositories.yaml

```diff
@@ -6,67 +6,54 @@ version: "2"
 
 # See https://mozilla.github.io/probe-scraper/#tag/library
 libraries:
-  - v1_name: glean-core
+  - library_name: glean
     description: Modern cross-platform telemetry (core library)
     notification_emails:
       - frank@mozilla.com
       - mdroettboom@mozilla.com
     url: https://github.com/mozilla/glean
-    branch: main
-    metrics_files:
-      - glean-core/metrics.yaml
-    ping_files:
-      - glean-core/pings.yaml
-    library_names:
-      - glean-core
-
-  - v1_name: glean-android
-    description: Modern cross-platform telemetry (Android-specific)
-    notification_emails:
-      - frank@mozilla.com
-      - mdroettboom@mozilla.com
-    url: https://github.com/mozilla/glean
-    branch: main
-    metrics_files:
-      - glean-core/android/metrics.yaml
-      - glean-core/metrics.yaml
-    ping_files:
-      - glean-core/pings.yaml
-    library_names:
-      - org.mozilla.components:service-glean
-
-  - v1_name: glean
-    description: Modern cross-platform telemetry (old)
-    deprecated: true
-    notification_emails:
-      - frank@mozilla.com
-      - mdroettboom@mozilla.com
-    url: https://github.com/mozilla/glean
-    branch: main
-    metrics_files:
-      - glean-core/android/metrics.yaml
-      - glean-core/metrics.yaml
-    ping_files:
-      - glean-core/pings.yaml
-    library_names: []
-
-  - v1_name: glean-deprecated
-    description: Modern cross-platform telemetry (old)
-    deprecated: true
-    notification_emails:
-      - frank@mozilla.com
-      - mdroettboom@mozilla.com
-    url: https://github.com/mozilla/glean
-    branch: main
-    metrics_files:
-      - glean-core/android/metrics.yaml
-      - glean-core/metrics.yaml
-    ping_files:
-      - glean-core/pings.yaml
-    library_names:
-      - org.mozilla.deprecated:glean
+    variants:
+      - v1_name: glean-core
+        description: Modern cross-platform telemetry (core library)
+        metrics_files:
+          - glean-core/metrics.yaml
+        ping_files:
+          - glean-core/pings.yaml
+        library_names:
+          - glean-core
+      - v1_name: glean-android
+        description: Modern cross-platform telemetry (Android-specific)
+        metrics_files:
+          - glean-core/android/metrics.yaml
```

Comment:
These metrics are available to `glean-android` but not `glean-core`.

If my mental model is correct, libraries should be grouped as variants if they share _any_ metric definitions. The set of shared metrics is a subset of all metrics for a library variant.

We will run into problems if a variant has shared metrics with _two_ libraries, but I don't see that happening.